Dockerfile

```diff
@@ -1,6 +1,6 @@
 ARG GDAL=ubuntu-small-3.3.3
 ARG PYTHON_VERSION=3.8
-FROM osgeo/gdal:${GDAL} AS gdal
```

Comment:
`${GDAL}` is defined in the `ARG GDAL` above and is meant to represent the image tag.

Reply:
https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact

Reply:
Also see: https://github.com/rasterio/rasterio/blob/72bdbc05b4670043fb81413e6ee718f06617b86c/Makefile#L30-L31

Reply:
My mistake - I meant to change the recipe for the  `make dockertest` command.  I'm having trouble building a Docker image, and thought I'd narrowed it down to this line in the makefile. I'll troubleshoot this some more and get back to you.

Reply:
Running `make dockertest` in a new conda environment returns the following error:

```
> make dockertest
docker build --build-arg GDAL= --build-arg PYTHON_VERSION= --target gdal -t rasterio:-py .
invalid argument "rasterio:-py" for "-t, --tag" flag: invalid reference format
See 'docker build --help'.
make: *** [Makefile:31: dockertestimage] Error 125
```
Which I think is understandable, but does this imply the requirement of GDAL to be previously installed on the system? Explicitly setting the following environment variables before running `make` helps:
```
> export PYTHON_VERSION=3.8
> export GDAL=ubuntu-small-3.3.3
> make dockertest
docker build --build-arg GDAL=ubuntu-small-3.3.3 --build-arg PYTHON_VERSION=3.8 --target gdal -t rasterio:ubuntu-small-3.3.3-py3.8 .
..

======================================= 2047 passed, 19 skipped, 1 deselected, 11 xfailed, 9 xpassed, 1653 warnings in 53.69s ========================================
```

I think the following lines should be added to the top of the Makefile to resolve this issue:
```
export PYTHON_VERSION=`python --version | cut -d' ' -f2`
export GDAL==ubuntu-small-3.3.3
```
This hard codes the value of GDAL though, which is not ideal.

Reply:
Here is a suggestion: https://stackoverflow.com/a/24264930